virtcontainers/kata_agent.go

```diff
@@ -1015,12 +1015,6 @@ func (k *kataAgent) constraintGRPCSpec(grpcSpec *grpc.Spec, passSeccomp bool) {
 		grpcSpec.Linux.Seccomp = nil
 	}
 
-	// Disable selinux
```

Comment:
@rhatdan Is this tested? Without this, the kata-agent applies the SelinuxLabel passed in the grpcSpec inside the guest, and since we dont have selinux suuport inside the guest, this was failing earlier.

Reply:
@amshinde what kind of `selinux support` is missing?

Reply:
@devimc Without this we were seeing this failure earlier with Kata `selinux label is specified in config, but selinux is disabled or not supported`, which essentially comes from https://github.com/opencontainers/runc/blob/4769cdf607b0d1cbd73e5407d9655da7511fe193/libcontainer/configs/validate/validator.go#L103

The cause of the failure is this : https://github.com/opencontainers/selinux/blob/b485497fbee5ba8d2d3e8f3bfea0d5f9f2d40e86/go-selinux/selinux_linux.go#L195

Its a failure in detecting an selinuxfs mount point.  
I think we would need to first enable selinux configs in kata guest kernel.

Reply:
Yes we should not pass the selinux label into the guest kernel.  For now, we only care about the process running the VM not used inside of the VM.

Reply:
Ok updated patch to remove SELinux label from oci spec before running container inside of VM.

Reply:
Thanks @rhatdan 
We do need to confine the container process inside the VM using selinux eventually, but this can be done later.
What I see is that we need two selinux labels, one for the hypervisor and one for the container process itself.

In case of runc, the selinux label passed in the OCI spec is applied to the container process itself. With this PR, we have a different behaviour as this label is applied to the hypervisor process instead, we can document this for now.
But to have expected behaviour, it looks like we need to have support for an additional selinux label for vm based runtimes in the spec.

Reply:
Using SELinux inside of the VM requires the VM to load SELinux policy.  We could potentially have a stripped down policy for containers within a VM, or another option would be to use the policy from the host system within the VM.  Anyways, yes we would need to create a new selinux type for the VM, or just use the types for the sandbox to apply to the VM and the types for each container to apply within the VM.  Bottom line we are a long way from this kind of support at this point.

Reply:
I'm still slightly unclear on this. If we're applying the generic SELinux policy to the hypervisor process (qemu, fc, etc), is that policy the same as that used for `runc`? What if we need to add support for a new hypervisor that needs additional SELinux privs beyond what a `runc` process would need?

- Would those privs be added to the generic policy (which would effectively give *too many* privs to non-VM containers like `runc`)?
- Would we need to create a new policy for all hypervisors at that point in time (so that *all* the VM-based containers had the extra privs, but non-VM containers didn't?)
- Would we need to create a new policy for that specific hypervisor at that point in time (so that *only* that hypervisor got the extra privs?

I think we should create some documentation on what type of SELinux support is being provided and what isn't. The question is where best to put it :smile: I suspect the [`docs`](https://github.com/kata-containers/documentation) repo might be the best place with a link to that from this repos top-level `README.md`.

Reply:
> I'm still slightly unclear on this. If we're applying the generic SELinux policy to the hypervisor process (qemu, fc, etc), is that policy the same as that used for `runc`? What if we need to add support for a new hypervisor that needs additional SELinux privs beyond what a `runc` process would need?
It will be different then what runc provides.  Runc needs ability to work with the host kernel and certain permissions, but in general can not work with /dev/kvm for instance.  When running a Virt Launching tool like qemu or Kata we will have to give it very different access.  But the bottom line from a security point of view is to guarantee separation between containers and the host.  We want to make sure that Container A can not interfere with Container B, and that Container A can not attack the host os in places like /var, /home, /root, /etc ...

The overlap between the policies will be that both can read/write/execute container_file_t content, and they will be separated by MCS Separation.  We need to make sure also that a runc/crun container/pod can not attack a KVM container/pod or vice versa.  

But bottom line, they both do very different things.
> 
>     * Would those privs be added to the generic policy (which would effectively give _too many_ privs to non-VM containers like `runc`)?
> 
Yes, I don't want runc containers to have the added allow rules of a KVM launcher, Similarly I would like to remove rules from a KVM Launcher that I might need to allow to runc/crun containers/pods.
>     * Would we need to create a new policy for all hypervisors at that point in time (so that _all_ the VM-based containers had the extra privs, but non-VM containers didn't?)
Yes
> 
>     * Would we need to create a new policy for that specific hypervisor at that point in time (so that _only_ that hypervisor got the extra privs?
> 
No, I will attempt to group all the hypervisors in the same policy.  Only issue would be if a particular hypervisor needed a grossly different and powerful permission to run.
> 
> I think we should create some documentation on what type of SELinux support is being provided and what isn't. The question is where best to put it smile I suspect the [`docs`](https://github.com/kata-containers/documentation) repo might be the best place with a link to that from this repos top-level `README.md`.

We could build out the github.com/containers/selinux README. which is where this would be best described.